controllers/clusteroperator_controller.go

```diff
@@ -54,14 +59,66 @@ type CloudOperatorReconciler struct {
 
 // Reconcile will process the cloud-controller-manager clusterOperator
 func (r *CloudOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	infraList := &configv1.InfrastructureList{}
+	if err := r.List(ctx, infraList); err != nil {
+		klog.Errorf("Unable to retrive list of Infrastructure objects: %v", err)
+		return ctrl.Result{}, err
+	}
```

Comment:
Shouldn't there only be one infrastructure we care about, I would expect us to get just that one rather than listing all of them and iterating through? I think this is optimising for a case that won't happen any time soon in OCP and  might lead to issues

Reply:
```suggestion
	infra := configv1.Infrastructure{}
	
	if err := r.Get(ctx, types.NamespacedName{Name: "cluster", Namespace: ""}, &infra); err != nil {
		klog.Errorf("Unable to retrive the infrastructure object: %v", err)
		return ctrl.Result{}, err
	}

	resources := getResources(infra)
	if err := r.sync(ctx, resources); err != nil {
		klog.Errorf("Unable to sync operands: %s", err)
	}
```

Reply:
All updated @Fedosin @JoelSpeed PTAL